### PR TITLE
management command to clone a domain

### PR DESCRIPTION
**Pull request by snopoke**
@dannyroberts 

cc @dimagi/scale-team for creating the ICDS SQL domain

### CODE REVIEW

File: corehq/apps/hqadmin/management/commands/clone_domain.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    def copy_products(self, old_domain, new_domain):
+        from corehq.apps.products.models import Product
+        from corehq.apps.programs.models import Program
+        
+        program_map = {}
+        programs = Program.by_domain(old_domain)
+        for program in programs:
+            old_id, new_id = self.save_couch_copy(program, new_domain)
+            program_map[old_id] = new_id
+
```
Comment (dimagimon) [inline corehq/apps/hqadmin/management/commands/clone_domain.py]: W293 blank line contains whitespace

File: corehq/apps/hqadmin/management/commands/clone_domain.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                copy_location_hierarchy(child, id_map)
+
+        locations = Location.root_locations(old_domain)
+        for location in locations:
+            copy_location_hierarchy(location, {})
+            
+    def copy_products(self, old_domain, new_domain):
+        from corehq.apps.products.models import Product
+        from corehq.apps.programs.models import Program
+
```
Comment (dimagimon) [inline corehq/apps/hqadmin/management/commands/clone_domain.py]: W293 blank line contains whitespace

File: corehq/apps/hqadmin/management/commands/clone_domain.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+            report_map = self.copy_ucr_data(existing_domain, new_domain)
+            self.copy_applications(existing_domain, new_domain, report_map)
+
+    def clone_domain_and_settings(self, old_domain, new_domain):
+        from corehq.apps.domain.models import Domain
+        new_domain_obj = Domain.get_by_name(new_domain)
+        if new_domain_obj:
+            if raw_input(
+                '{} domain already exists. Do you still want to continue? [y/n]'.format(new_domain)
+                ).lower() == 'y':
```
Comment (dimagimon) [inline corehq/apps/hqadmin/management/commands/clone_domain.py]: E125 continuation line with same indent as next logical line
E123 closing bracket does not match indentation of opening bracket's line

File: corehq/apps/hqadmin/management/commands/clone_domain.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+            children = location.children
+            old_id, new_id = self.save_couch_copy(location, new_domain)
+            id_map[old_id] = new_id
+            for child in children:
+                copy_location_hierarchy(child, id_map)
+
+        locations = Location.root_locations(old_domain)
+        for location in locations:
+            copy_location_hierarchy(location, {})
+
```
Comment (dimagimon) [inline corehq/apps/hqadmin/management/commands/clone_domain.py]: W293 blank line contains whitespace

**Comment by czue** [comment]
this looks great to me. pretty interesting how un-generalizable the whole process was. also pretty scary how much magic happens syncing the couch models to sql in the save function (was surprised not to see any references to sql products and locations until I realized that would all just happen)

**Pull request merged**